### PR TITLE
SPU/GUI: Add MFC Delay Command checkbox to GUI, Add one more MFC Shuffle setting

### PR DESCRIPTION
**Pull request by elad335**
* Add "MFC Delay Commands" in Advanced tab, when set sets "MFC Commands Shuffling Limit" to 1.
* Add "MFC Commands Shuffling In Steps" setting in config.yml, with this there's another layer of randomness in MFC: you never know when the command will be executed exactly. Before you could only specify random ordering and delaying with predictable timeout. When combined with timeout the random execution will occur each specified time, but which commands will be executed at each period is random.
* Another fixup after #8514 regarding the new settings.

### CODE REVIEW

File: rpcs3/rpcs3qt/tooltips.h [hunk 1]
```diff
@@ -85,6 +85,7 @@ class Tooltips : public QObject
 		const QString accurate_llvm_dfma           = tr("Provides extra accuracy on FMA instructions at the cost of performance.\nWhile disabling it might give a decent performance boost if your CPU doesn't support FMA, it may also introduce subtle bugs that otherwise do not occur.\nYou can't disable it if your CPU supports FMA.");
 		const QString accurate_vector_nan          = tr("Forces the floating point NaN (Not A Number) values outputted from PPU vector instructions to be accurate to the real hardware. (0x7FC00000)");
 		const QString accurate_rsx_access          = tr("Forces RSX pauses on SPU MFC_GETLLAR and SPU MFC_PUTLLUC operations.");
+		const QString mfc_delay_command            = tr("Do not execute immediately an MFC command, wait for atleast 2 pending commands to execute them all in a random order.\nMust be used with either SPU interpreters currently.\nSeverely degrades performance!If unsure, don't use this option.");
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/tooltips.h:88]: Grammar

File: rpcs3/rpcs3qt/settings_dialog.cpp [hunk 2]
```diff
@@ -1027,6 +1027,14 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 	m_emu_settings->EnhanceCheckBox(ui->disableOnDiskShaderCache, emu_settings_type::DisableOnDiskShaderCache);
 	SubscribeTooltip(ui->disableOnDiskShaderCache, tooltips.settings.disable_on_disk_shader_cache);
 
+	ui->mfcDelayCommand->setChecked(m_emu_settings->GetSetting(emu_settings_type::MFCCommandsShuffling) == "1");
+	SubscribeTooltip(ui->mfcDelayCommand, tooltips.settings.mfc_delay_command);
+	connect(ui->mfcDelayCommand, &QCheckBox::stateChanged, [&](int val)
+	{
+		const std::string str = val != 0 ? "1" : "0";
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/settings_dialog.cpp:1034]:
```suggestion
		const std::string str = val != Qt::Unchecked ? "1" : "0";
```

Comment (elad335) [reply rpcs3/rpcs3qt/settings_dialog.cpp]: Fixed.

File: rpcs3/rpcs3qt/tooltips.h (hunk 1 above)
Comment (elad335) [reply rpcs3/rpcs3qt/tooltips.h]: Fixed.

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 3]
```diff
@@ -3106,6 +3112,14 @@ void spu_thread::do_mfc(bool can_escape)
 		{
 			break;
 		}
+
+		if (!must_finish && g_cfg.core.mfc_shuffling_in_steps)
+		{
+			// Exit early, not all pending commands have to be executed at a single iteration
+			// Update last timestamp so the next MFC timeout check will use the curent time
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/SPUThread.cpp:3119]:
```suggestion
			// Update last timestamp so the next MFC timeout check will use the current time
```

File: rpcs3/rpcs3qt/tooltips.h [hunk 4]
```diff
@@ -85,6 +85,7 @@ class Tooltips : public QObject
 		const QString accurate_llvm_dfma           = tr("Provides extra accuracy on FMA instructions at the cost of performance.\nWhile disabling it might give a decent performance boost if your CPU doesn't support FMA, it may also introduce subtle bugs that otherwise do not occur.\nYou can't disable it if your CPU supports FMA.");
 		const QString accurate_vector_nan          = tr("Forces the floating point NaN (Not A Number) values outputted from PPU vector instructions to be accurate to the real hardware. (0x7FC00000)");
 		const QString accurate_rsx_access          = tr("Forces RSX pauses on SPU MFC_GETLLAR and SPU MFC_PUTLLUC operations.");
+		const QString mfc_delay_command            = tr("Forces delaying any odd MFC command, waits for atleast 2 pending commands to execute them in a random order.\nMust be used with either SPU interpreters currently.\nSeverely degrades performance! If unsure, don't use this option.");
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/tooltips.h:88]:
```suggestion
		const QString mfc_delay_command            = tr("Forces delaying any odd MFC command, waits for at least 2 pending commands to execute them in a random order.\nMust be used with either SPU interpreters currently.\nSeverely degrades performance! If unsure, don't use this option.");
```

File: rpcs3/Emu/Cell/SPUThread.cpp (hunk 3 above)
Comment (elad335) [reply rpcs3/Emu/Cell/SPUThread.cpp]: Fixed.

File: rpcs3/rpcs3qt/tooltips.h (hunk 4 above)
Comment (elad335) [reply rpcs3/rpcs3qt/tooltips.h]: Fixed.

**Pull request merged**